tests/tasks/test_control_flow.py

```diff
@@ -225,6 +225,57 @@ def false_branch():
         assert state.result[merge_task].result == 0
 
 
+def test_back_to_back_switches():
+    # regression test for https://github.com/PrefectHQ/prefect/issues/2017
+    log = []
+
+    @prefect.task()
+    def is_a_enabled():
+        return True
+
+    @prefect.task()
+    def is_b_enabled():
+        return False
+
+    @prefect.task()
+    def is_c_enabled():
+        return True
+
+    @prefect.task()
+    def start():
+        nonlocal log
```

Comment:
You shouldn't need to declare this since you aren't creating it here, only altering it.